zebra/if_netlink.c

```diff
@@ -2127,6 +1959,16 @@ int netlink_link_change(struct nlmsghdr *h, ns_id_t ns_id, int startup)
 			/* Update link. */
 			zebra_if_update_link(ifp, link_ifindex, link_nsid);
 
+			/*
+			 * Just set the @link/lower-device ifindex. During
+			 * nldump interfaces are not ordered in any fashion so
+			 * we may end up getting upper devices before lower
+			 * devices. We will setup the real linkage once the dump
+			 * is complete.
+			 */
+			zif = (struct zebra_if *)ifp->info;
+			zif->link_ifindex = link_ifindex;
+
```

Comment:
`zebra_if_update_link` above does exactly the same thing. This code is not needed.

Reply:
Well the zebra_if_update_link code ignores the value setting if the link type is VETH?  I am hesitant to blindly change this without further investigation.  I'll file an issue and we can come back and fix this in the future imo.

Reply:
#13533